aioaseko/mobile.py

```diff
@@ -38,17 +43,26 @@ def __init__(
         username: str | None = None,
         password: str | None = None,
         access_token: str | None = None,
+        access_token_expiration: int | None = None,
+        refresh_token: str | None = None,
     ) -> None:
         """Init Aseko account."""
         self._session = session
         self._username = username
         self._password = password
         self._access_token = access_token
+        self._access_token_expiration = access_token_expiration
+        self._refresh_token = refresh_token
 
     @property
-    def access_token(self) -> str | None:
-        """Return access token."""
-        return self._access_token
+    def refresh_token(self) -> str | None:
+        """Return refresh token."""
+        return self._refresh_token
+
+    @property
+    def access_token_expiration(self) -> int | None:
```

Comment:
The requests and access tokens are created and managed by the library. The user of the library doesn't need to worry about refreshing access tokens so I don't think it's necessary to expose the `access_token_expiration` property and the `access_token` and `retrieve_tokens` methods to the user. We could also remove the `access_token` arguments in `__init__`. In practice this will not be used anyway because access tokens are only valid for 15 minutes so storing it outside the class instance is not useful and we can always create a new one with the refresh token. With these changes we can simplify the code and make the API easier for the user.

Reply:
Thanks, I have applied all the suggested changes & added few fixes as I discovered during the testing, that our logic was breaking when there was deployment/outage of the Aseko Pool Service & the tokens were no longer valid.